logbook-json/src/test/java/org/zalando/logbook/json/JsonBodyFiltersTest.java

```diff
@@ -93,6 +97,37 @@ void shouldFilterNullProperty() {
         assertThat(actual, is("{\"foo\":\"XXX\",\"bar\":\"public\"}"));
     }
 
+    @Test
+    void shouldFilterMatchingNullPropertyOnly() {
+        final BodyFilter unit = replaceJsonStringProperty(singleton("foo"), "XXX");
+
+        final String actual = unit.filter(contentType, "{\"foo\":null,\"bar\":null}");
+
+        assertThat(actual, is("{\"foo\":\"XXX\",\"bar\":null}"));
+    }
+
+    @Test
+    void shouldFilterUsingGivenSetSemantics() {
+        final SortedSet<String> properties = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+        properties.add("FOO");
+
+        final BodyFilter unit = replaceJsonStringProperty(properties, "XXX");
+
+        final String actual = unit.filter(contentType, "{\"foo\":null,\"bar\":null}");
+
+        assertThat(actual, is("{\"foo\":\"XXX\",\"bar\":null}"));
+    }
+
+    @Test
+    void shouldFilterPrimitives() {
+        final BodyFilter unit = replacePrimitiveJsonProperty(
+                Arrays.asList("foo", "bar", "baz")::contains, "XXX");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Arrays.asList' due to existing static import 'java.util.Arrays.asList'](https://app.codacy.com/app/whiskeysierra/logbook/pullRequest?prid=4602133)